### ISSUE TITLE
About nested menuitems in a menu with roving tabindex

### ISSUE THREAD

**Issue by skmanohar** [dequelabs/axe-core#4246]
### Product

axe-core

### Question

Nested menus with roving tabindex triggers `aria-required-children` when implemented like this:

### Implementation 1
`tabindex=-1` is set on a parent of the `menuitem`
```html
<ul role=menu>
  <li role=none>
    <custom-component>
      <button role=menuitem>Menu Item 1</button>
     </custom-component>
  </li>
  <li role=none>
    <custom-component tabindex=-1>
      <button role=menuitem>Menu Item 2</button>
     </custom-component>
  </li>
</ul>
```
### Implementation 2
`tabindex=-1` is set on the `menuitem`; this does not trigger `aria-required-children`:
```html
<ul role=menu>
  <li role=none>
    <custom-component>
      <button role=menuitem>Menu Item 1</button>
     </custom-component>
  </li>
  <li role=none>
    <custom-component>
      <button role=menuitem tabindex=-1>Menu Item 2</button>
     </custom-component>
  </li>
</ul>
```

I suspect that axe-core's callout for Implementation1 might be a false positive. Wanted to confirm.

axe-core 4.8.2

**Comment by skmanohar** [comment]
Adding a stackblitz: https://stackblitz.com/edit/web-platform-ybppvz?file=index.html

**Comment by straker** [comment]
Thanks for the issue. The reason axe is reporting an issue is because of [presentation-role-conflict](https://www.w3.org/TR/wai-aria-1.2/#conflict_resolution_presentation_none). When the `custom-component` has a `tabindex` it no longer resolves the element as having a presentational role (`none` to be more specific) and instead assigns the element to have the [generic role](https://www.w3.org/TR/wai-aria-1.2/#generic). This causes the menu to have a non-menuitem as a direct child and thus causes the violation.

**Comment by skmanohar** [comment]
Understood, thank you for the detailed clarification!
